tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -288,6 +288,8 @@ static PyObject * THPVariable_randint(PyObject* self_, PyObject* args, PyObject*
   END_HANDLE_TH_ERRORS
 }
 
+// bound here bc it is more performant to construct the tensor while recursing through
+// python object
```

Comment:
This isn't the main reason, right? `data` is pretty dynamic (it can be any type of array like object -- list, tuple, np array, scalar) and we are unable to represent that in native_functions

Reply:
yea fair. it was just one reason stated. i will update it.